tel2puml/pv_to_puml.py

```diff
@@ -255,26 +255,90 @@ def pv_events_from_folder_to_puml_file(
 
 def pv_streams_to_puml_files(
     pv_streams: Generator[
-        tuple[str, Generator[Generator[PVEvent, Any, None], Any, None]],
+        tuple[str, Generator[Iterable[PVEvent], Any, None]],
         Any,
         None,
     ],
-    file_directory: str = ".",
+    output_file_directory: str = ".",
 ) -> None:
     """
     Function to convert and save a stream of PVEvents to puml files.
 
     :param pv_streams: Generator of tuples of job_name to generator of
     generators of PVEvents grouped by job_name, then job_id.
-    :type pv_streams: `Generator`[`tuple`[`str`, `Generator`[`Generator`
-    [:class:`PVEvent`, `Any`, `None`], `Any`, `None`]], `Any`, `None`]
-    :param file_directory: The file directory to store puml files. Defaults to
-    "."
-    :type file_directory: `str`
+    :type pv_streams: `Generator`[`tuple`[`str`, `Generator`
+    [`Iterable`[:class`PVEvent`], `Any`, `None`]], `Any`, `None`]
+    :param output_file_directory: The file directory to store puml files.
+    Defaults to "."
+    :type output_file_directory: `str`
     """
     for job_name, job_event_gen in pv_streams:
-        puml_file_path = os.path.join(file_directory, f"{job_name}.puml")
+        job_name = job_name.replace(" ", "_")
+        puml_file_path = os.path.join(
+            output_file_directory, f"{job_name}.puml"
+        )
         pv_to_puml_file(
             job_event_gen,
             puml_file_path=puml_file_path,
         )
+
+
+def pv_files_to_pv_streams(
+    file_directory: Optional[str] = None,
```

Comment:
Think we should only have file_list here and we will deal with converting file_directory to file_paths in the cli bit